### PR TITLE
MM-30355: Cypress/E2E: Automate backlogs - Account Settings > Username (6 test cases)

### PR DESCRIPTION
**Pull request by catalintomai**
#### Summary
Cypress/E2E: Automate backlogs - Account Settings > Username (6 test cases)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30355

### CODE REVIEW

File: e2e/cypress/integration/account_settings/general/username_spec.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        cy.get('#saveSetting').click();
+
+        // * Check if element is present and contains expected text values
+        cy.get('#clientError').should('be.visible').should('contain', 'Username must begin with a letter, and contain between 3 to 22 lowercase characters made up of numbers, letters, and the symbols \'.\', \'-\', and \'_\'.');
+
+        // # Click "x" button to close Account Settings modal
+        cy.get('#accountSettingsHeader > .close').click();
+    });
+
+    //Note: the test case mentions an error message that could not be found in the codebase: "This username is already taken. Please choose another." Added a note to the testcase to update the error message.
```
Comment (saturninoabril) [inline e2e/cypress/integration/account_settings/general/username_spec.js:74]: Thanks for checking! Looks like it was recently changed by https://github.com/mattermost/mattermost-server/pull/15563. I've update the test case to reflect the new error message: "Unable to find the existing account to update.".
```suggestion
```

File: e2e/cypress/integration/account_settings/general/username_spec.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            cy.get('#suggestionList').should('be.visible');
+
+            // # Type user name
+            cy.get('#post_textbox').type(newTempUserName);
+            cy.get('#post_textbox').type('{enter}{enter}');
+
+            // # Check that the user name has been posted
+            cy.getLastPostId().then((postId) => {
+                cy.get(`#postMessageText_${postId}`).should('contain', newTempUserName);
+            });
```
Comment (saturninoabril) [inline e2e/cypress/integration/account_settings/general/username_spec.js:147]: I think it's fine to at-mention self but still missing the last step:
- "4. Click "@" beside the search box to view recent mentions" then verify "Recent mentions button searches for updated username"

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks Catalin! Overall looks good :100:, except for one missing step/verification on MM-T2053.

**Review by saturninoabril — APPROVED** [review]
Thanks @catalintomai! LGTM.

**Comment by DHaussermann** [comment]
@catalintomai  or @saturninoabril Sorry for the delay in testing this. Seem master can not merge into the branch automatically anymore. 

All tests work and run green. Could one of you please review this merge issue ☝️ and advise if this need another test pass after?

**Comment by catalintomai** [comment]
> @catalintomai or @saturninoabril Sorry for the delay in testing this. Seem master can not merge into the branch automatically anymore.
> 
> All tests work and run green. Could one of you please review this merge issue ☝️ and advise if this need another test pass after?

@DHaussermann , I merged and hardened a bit the last test, please give it another test pass to make sure.

**Review by DHaussermann — APPROVED** [review]
Test and passed (after master merge)
- Test match TM4J intent
- All test run Green
- Updated TM4J Cypress status accordingly

LGTM!

**Pull request merged**